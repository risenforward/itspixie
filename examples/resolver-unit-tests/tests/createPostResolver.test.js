const resolvers = require('../src/resolvers')
const { getPrismaTestInstance } = require('./getPrismaTestInstance')

afterEach(async () => {
  await getPrismaTestInstance().mutation.deleteManyPosts({})
})

test('creating draft works', async () => {
  const parent = {}
  const args = {
    data: {
      content: 'some text',
      status: 'DRAFT',
      title: 'New post',
    },
  }
  const ctx = {
    db: getPrismaTestInstance(),
  }

  const info = `{ content, status, title }`

  expect(
    await resolvers.Mutation.createPost(parent, args, ctx, info),
  ).toMatchSnapshot()
})

test('creating publish fails', async () => {
  try {
    const parent = {}
    const args = {
      data: {
        content: 'some text',
        status: 'PUBLISH',
        title: 'New post',
      },
    }
    const ctx = {
      db: getPrismaTestInstance(),
    }

    const info = `{ id }`
    const result = await resolvers.Mutation.createPost(parent, args, ctx, info)
    expect(0).toBe(1)
  } catch (e) {
    expect(e.toString()).toMatchSnapshot()
  }
})
