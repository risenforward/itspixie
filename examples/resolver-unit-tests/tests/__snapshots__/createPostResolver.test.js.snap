// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creating draft works 1`] = `
Object {
  "content": "some text",
  "status": "DRAFT",
  "title": "New post",
}
`;

exports[`creating publish fails 1`] = `"Error: Can't create published posts!"`;
